File: app/src/main/java/com/example/resourcesapp/ui/projectIdeas/ProjectIdeasFragment.java
package com.example.resourcesapp.ui.projectIdeas;

import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.resourcesapp.R;

public class ProjectIdeasFragment extends Fragment {

    private ProjectIdeasViewModel mViewModel;

    public static ProjectIdeasFragment newInstance() {
        return new ProjectIdeasFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_project_ideas, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(ProjectIdeasViewModel.class);
        // TODO: Use the ViewModel
    }

}

File: app/src/main/java/com/example/resourcesapp/ui/projectIdeas/ProjectIdeasFragment.kt
package com.example.resourcesapp.ui.projectIdeas

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.resourcesapp.R

class ProjectIdeasFragment : Fragment() {
    private var mViewModel: ProjectIdeasViewModel? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_project_ideas, container, false)
    }

    @Deprecated("Deprecated in Java")
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        mViewModel = ViewModelProvider(this).get(ProjectIdeasViewModel::class.java)
        // TODO: Use the ViewModel
    }

    companion object {
        fun newInstance(): ProjectIdeasFragment {
            return ProjectIdeasFragment()
        }
    }
}